hibernate-rx-core/src/main/java/org/hibernate/rx/persister/entity/impl/HiLoRxOptimizer.java

```diff
@@ -0,0 +1,56 @@
+package org.hibernate.rx.persister.entity.impl;
+
+import org.hibernate.AssertionFailure;
+import org.hibernate.id.IdentifierGeneratorHelper;
+import org.hibernate.id.enhanced.HiLoOptimizer;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+public class HiLoRxOptimizer extends HiLoOptimizer implements RxOptimizer<Long> {
+
+	public HiLoRxOptimizer(Class<?> integerType, int incrementSize) {
+		super( integerType, incrementSize );
+	}
+
+	@Override
+	public synchronized CompletionStage<Optional<Long>> generate(RxAccessCallback callback) {
```

Comment:
This looks correct to me but I'm wondering if we could test it under stress to have some confidence that it's actually thread safe.

This is an example of what we did for OGM: https://github.com/hibernate/hibernate-ogm/blob/master/core/src/test/java/org/hibernate/ogm/backendtck/id/SequenceNextValueGenerationTest.java